src/utils.js

```diff
@@ -226,6 +226,11 @@ function truncateString(str, maxStrLength = 10) {
   return `${str.slice(0, maxStrLength)}...`;
 }
 
+function urlValidation(url) {
+  const regex = new RegExp('(https?://)?([\\da-z.-]+)\\.([a-z.]{2,6})[/\\w .-]*/?');
```

Comment:
its not a deal breaker but sometimes using a system library does a better job than a regex, example: https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url

Reply:
could somethine like this work? idk if https is our main requirements, plus being a valid URI then this one should also work https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

```
url = 'http://test'
'http://test'


try {
    new URL(url); if (url.startsWith('https') === false) { throw new TypeError('failed since not https'); }
} catch(e) { console.error('failed with error ' , e) }
VM1144:3 failed with error  TypeError: failed since not https
    at <anonymous>:2:66
```

Reply:
but if there are other requirements beyond met by above, then yeah you may have to check

Reply:
new URL() will catch 'basic' error in urls, and the second check catches http vs https